### PR TITLE
[quant][graphmode][fx] Add fbgemm backend_config_dict

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64288
* #64135

Summary:
This is just to setup the file structure and unblock experimentation.
The format for backend_config_dict will change in the future

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30699457](https://our.internmc.facebook.com/intern/diff/D30699457)

### CODE REVIEW

File: torch/quantization/fx/backend_config_dict/fbgemm.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+from ..quantization_patterns import *
+from ..pattern_utils import get_default_quant_patterns
+
+def get_fbgemm_backend_config_dict():
+    """ Get the backend config dictionary for fbgemm backend
+    NOTE: Current api is not final, it's just to unblock experimentation for new backends
+    """
+    # TODO: add output_activation_post_process_map
+    return {
+        "quant_patterns": get_default_quant_patterns()
```
Comment (vkuzo) [inline torch/quantization/fx/backend_config_dict/fbgemm.py:11]: might be good to clarify further that this API will definitely change, and it's 100% unsafe for customers to depend on this in any way until further notice.

File: torch/quantization/fx/backend_config_dict/fbgemm.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import torch
+from ..quantization_patterns import *
+from ..pattern_utils import get_default_quant_patterns
+
+def get_fbgemm_backend_config_dict():
+    """ Get the backend config dictionary for fbgemm backend
+    NOTE: Current api is not final, it's just to unblock experimentation for new backends
```
Comment (vkuzo) [inline torch/quantization/fx/backend_config_dict/fbgemm.py:7]: can we also add some checking that the values are valid, or add a TODO to do it before users start using the API?

**Review by vkuzo — APPROVED** [review]

File: torch/quantization/fx/backend_config_dict/fbgemm.py (hunk 2 above)
Comment (jerryzh168) [reply torch/quantization/fx/backend_config_dict/fbgemm.py:7]: you mean `validate_backend_config_dict`? yeah we have that in __init__.py

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30699457).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30699457).

**Pull request closed without merging**